Mathlib/CategoryTheory/MorphismProperty/Limits.lean

```diff
@@ -388,38 +396,45 @@ instance : (W.limitsOfShape J).RespectsIso :=
     rw [← hg]
     exact ⟨_, _, _, _, h₁', _, _, hf⟩)
 
-variable {J} in
+variable {W J} in
 lemma limitsOfShape_limMap {X Y : J ⥤ C}
     (f : X ⟶ Y) [HasLimit X] [HasLimit Y] (hf : W.functorCategory _ f) :
     W.limitsOfShape J (limMap f) :=
   ⟨_, _, _, _, limit.isLimit X, _, _, hf⟩
 
 /-- The property that a morphism property `W` is stable under limits
 indexed by a category `J`. -/
-def IsStableUnderLimitsOfShape : Prop :=
-  ∀ (X₁ X₂ : J ⥤ C) (c₁ : Cone X₁) (c₂ : Cone X₂)
-    (_ : IsLimit c₁) (h₂ : IsLimit c₂) (f : X₁ ⟶ X₂) (_ : W.functorCategory J f),
-      W (h₂.lift (Cone.mk _ (c₁.π ≫ f)))
+class IsStableUnderLimitsOfShape : Prop where
+  condition (X₁ X₂ : J ⥤ C) (c₁ : Cone X₁) (c₂ : Cone X₂)
```

Comment:
Should all of these arguments be explicit?

Reply:
In the applications, I find it slightly more convenient.